src/java/org/apache/cassandra/db/commitlog/CommitLogReplayer.java

```diff
@@ -130,7 +130,16 @@ public static CommitLogReplayer construct(CommitLog commitLog, UUID localHostId)
                 }
             }
 
-            IntervalSet<CommitLogPosition> filter = persistedIntervals(cfs.getLiveSSTables(), truncatedAt, localHostId);
+            IntervalSet<CommitLogPosition> filter;
+            if (!cfs.memtableWritesAreDurable())
+            {
+                filter = persistedIntervals(cfs.getLiveSSTables(), truncatedAt, localHostId);
+            }
+            else
+            {
+                // everything is persisted and restored by the memtable itself
+                filter = new IntervalSet<>(CommitLogPosition.NONE, CommitLog.instance.getCurrentPosition());
```

Comment:
When memtableWritesAreDurable, restore to a point in time does not get correct intervals with this IntervalSet

Reply:
Added functionality to specify snapshot commit log position, so that user can specify that when a persistent memtable is recovered with a known start position, and a default of no persisted interval if that is not specified. This should cover both the initial replay-all-commitlogs solution and a future option with memtable snapshots.